wadas/domain/animal_detection_mode.py

```diff
@@ -46,11 +47,44 @@ def run(self):
                 ):
                     logger.info("Instantiating FTPS server...")
                     self.ftp_thread = FTPsServer.ftps_server.run()
+        logger.info("Ready for video stream from Camera(s)...")
 
-        self.start_actuator_server()
+    def _detect(self, cur_image_path):
+        results, detected_img_path = self.ai_model.process_image(cur_image_path, True)
+        self.last_detection = detected_img_path
+        return results, detected_img_path
+
+    def _format_classified_animals_string(self, classified_animals):
+        # Prepare a list of classified animals to print in UI
+        if self.last_classified_animals_str:
+            self.last_classified_animals_str = ""
+        for animal in classified_animals:
+            last = animal["classification"][0]
+            if not self.last_classified_animals_str:
+                self.last_classified_animals_str = self.last_classified_animals_str + last
+            else:
+                self.last_classified_animals_str = self.last_classified_animals_str + ", " + last
+
+    def _classify(self, cur_image_path, detection_results):
+        # Classify if detection has identified animals
+        if len(detection_results["detections"].xyxy) > 0:
```

Comment:
if len(detection_results["detections"].xyxy):

or (but maybe with pandas or something like that doesn't work. Again, to be checked):

if detection_results["detections"].xyxy: